### PR TITLE
test(react): enhance use-focus coverage

### LINKED ISSUE #5717: Enhance Test Coverage for `useFocus` in `@yamada-ui/react`
package: @yamada-ui/react
group: useFocus

### Description
Test coverage for `useFocus` in `@yamada-ui/react` is less than 95%. Please increase it to at least 95%.

### Target files
- [packages/react/src/hooks/use-focus/index.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/react/src/hooks/use-focus/index.ts)
  - [L68](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/react/src/hooks/use-focus/index.ts#L68)
  - [L126](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/react/src/hooks/use-focus/index.ts#L126)

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

### PR DESCRIPTION
**Pull request by hanityx**
## Summary

This PR improves test coverage for `useFocus` hooks in `@yamada-ui/react`.

I focused on branch coverage for both `useFocusOnShow` and `useFocusOnPointerDown`, including previously uncovered edge cases.

### Added coverage for
- focus fallback when no focusable descendant exists
- direct `HTMLElement` usage for `refOrEl`
- early-return paths (`shouldFocus: false`, unresolved target, active element already inside target)
- pointer-down behavior when target is already active
- fallback behavior when `elements` is omitted
- mixed `elements` input (`HTMLElement` and `null`)
- non-Safari guard path

### Notes
- test-only changes (no production runtime logic changed)
- updated test typing to use `focusTarget` consistently
- stabilized utility mocking and test cleanup paths for reliability

## Verification

- `pnpm -C packages/react test -- --run src/hooks/use-focus/index.test.tsx`
- `pnpm -C packages/react test -- --run --project=jsdom --coverage --coverage.provider=v8 src/hooks/use-focus/index.test.tsx`

Result:
- 13 tests passed
- `src/hooks/use-focus/index.ts` coverage: 100% (statements/branches/functions/lines)

Closes #5717

### CODE REVIEW

**Comment by hanityx** [comment]
Follow-up update after local review pass:

- replaced utils namespace spying with module-level vi.mock("../../utils") for getFirstFocusableElement
- tightened async null-target assertion with waitFor
- stabilized mock lifecycle (mockReset) between tests
- kept behavioral assertions focused on DOM focus state

Local verification:
- pnpm -C packages/react test -- --run src/hooks/use-focus/index.test.tsx (13/13)
- use-focus/index.ts coverage remains 100% for statements/branches/functions/lines

Also, both Vercel checks on this fork PR show "Authorization required to deploy" (external integration auth), not test/lint failures.

File: packages/react/src/hooks/use-focus/index.test.tsx [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
 import type { UseFocusOnMouseDownProps, UseFocusOnShowProps } from "./"
 import { act, fireEvent, render, waitFor } from "#test"
 import { useRef } from "react"
+import { getFirstFocusableElement } from "../../utils"
 import { useFocusOnPointerDown, useFocusOnShow } from "./"
 
+const mockState = vi.hoisted(() => {
+  return {
+    defaultGetFirstFocusableElement:
+      undefined as unknown as Utils.getFirstFocusableElement,
```
Comment (hirotomoyamada) [inline packages/react/src/hooks/use-focus/index.test.tsx]: Change `undefined as unknown as Utils.getFirstFocusableElement` to `undefined as unknown as typeof getFirstFocusableElement`.

File: packages/react/src/hooks/use-focus/index.test.tsx [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
+  }
+})
+
+vi.mock("../../utils", async (importOriginal) => {
+  const actual = (await importOriginal()) as typeof Utils
```
Comment (hirotomoyamada) [inline packages/react/src/hooks/use-focus/index.test.tsx]: Change `const actual = (await importOriginal()) as typeof Utils` to `const actual = await importOriginal<typeof Utils>()`.

**Review by hirotomoyamada — CHANGES_REQUESTED** [review]
Please correct the two comments.

File: packages/react/src/hooks/use-focus/index.test.tsx (hunk 1 above)
Comment (hanityx) [reply packages/react/src/hooks/use-focus/index.test.tsx]: Done — changed to `typeof getFirstFocusableElement`.

File: packages/react/src/hooks/use-focus/index.test.tsx (hunk 2 above)
Comment (hanityx) [reply packages/react/src/hooks/use-focus/index.test.tsx]: Updated to `await importOriginal<typeof Utils>()`. Thanks!

**Comment by hanityx** [comment]
Both fixed. Could you take another look?

**Review by hirotomoyamada — APPROVED** [review]
Thanks!
